src/lib.rs

```diff
@@ -110,63 +124,107 @@ impl<L: LogLevel> Verbosity<L> {
+        self.filter() == Filter::Off
     }
 
-    fn verbosity(&self) -> u8 {
-        let default_verbosity = level_value(L::default());
-        let verbosity = default_verbosity as i16 - self.quiet as i16 + self.verbose as i16;
-        verbosity.clamp(0, u8::MAX as i16) as u8
+    /// Get the filter level after applying `--verbose` and `--quiet`.
+    pub fn filter(&self) -> Filter {
+        L::default().with_offset(self.verbose as i16 - self.quiet as i16)
     }
 }
 
-fn level_value(level: Option<Level>) -> u8 {
-    match level {
-        None => 0,
-        Some(Level::Error) => 1,
-        Some(Level::Warn) => 2,
-        Some(Level::Info) => 3,
-        Some(Level::Debug) => 4,
-        Some(Level::Trace) => 5,
-    }
+/// A representation of the log level filter.
+///
+/// Used to calculate the log level and filter.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum Filter {
+    Off,
+    Error,
+    Warn,
+    Info,
+    Debug,
+    Trace,
 }
 
-fn level_enum(verbosity: u8) -> Option<Level> {
-    match verbosity {
-        0 => None,
-        1 => Some(Level::Error),
-        2 => Some(Level::Warn),
-        3 => Some(Level::Info),
-        4 => Some(Level::Debug),
-        5..=u8::MAX => Some(Level::Trace),
+impl Filter {
+    /// Apply an offset to the filter level.
+    fn with_offset(&self, offset: i16) -> Filter {
+        let value = self.as_usize() as i16 + offset;
+        const MAX_LEVEL: i16 = 5;
+        Self::from_usize(value.clamp(0, MAX_LEVEL) as usize)
     }
-}
 
-use std::fmt;
+    /// Convert the filter to a usize for arithmetic.
+    ///
+    /// usize avoids negative values (and is used in the log crate).
+    fn as_usize(&self) -> usize {
+        match self {
+            Filter::Off => 0,
+            Filter::Error => 1,
+            Filter::Warn => 2,
+            Filter::Info => 3,
+            Filter::Debug => 4,
+            Filter::Trace => 5,
+        }
+    }
+
+    /// Convert a usize back to a filter.
+    fn from_usize(value: usize) -> Self {
+        match value {
+            0 => Filter::Off,
+            1 => Filter::Error,
+            2 => Filter::Warn,
+            3 => Filter::Info,
+            4 => Filter::Debug,
+            _ => Filter::Trace,
+        }
+    }
+}
 
 impl<L: LogLevel> fmt::Display for Verbosity<L> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}", self.verbosity())
+        match self.filter() {
+            Filter::Off => write!(f, "off"),
+            Filter::Error => write!(f, "error"),
+            Filter::Warn => write!(f, "warn"),
+            Filter::Info => write!(f, "info"),
+            Filter::Debug => write!(f, "debug"),
+            Filter::Trace => write!(f, "trace"),
+        }
     }
 }
 
 /// Customize the default log-level and associated help
 pub trait LogLevel {
     /// Base-line level before applying `--verbose` and `--quiet`
-    fn default() -> Option<Level>;
+    fn default() -> Filter;
```

Comment:
Would it make sense to rename this to `filter()` to avoid confusion with `Default::default()`, and app implementations of `LogLevel` that might wish to implement both traits for their apps?

Reply:
Good point about `Default::default`.  However, the name should convey that this is the starting point.  We could call it `default_filter`.

Reply:
Changed in #125